_posts/2022-10-28-multiple-data-source.md

```diff
@@ -0,0 +1,97 @@
+
+###What can you do with multi data source experimental launch?
+
+The first phase of multiple data source will provide OpenSearch Dashboards ability to query data from multiple OpenSearch endpoints with compatible versions, with this feature you can:
+
+* Add/remove the data sources which is compatible with Elasticsearch DSL, e.g. AWS OpenSearch Domains, AWS OpenSearch Serverless collections, and customer self-managed OpenSearch clusters
+* Create visualizations that renders data from different index patterns to compare time-series data in the same dashboard
+
+![Use case by persona](/assets/media/blog-images/2022-10-28-multiple-data-source/usecase-by-persona.png){: .img-fluid}
+
+
+###How does multi data source feature work?
+
+
+**High-level architecture diagram**
+
+![Legacy architecture](/assets/media/blog-images/2022-10-28-multiple-data-source/legacy.png){: .img-fluid}
+![Evolution architecture](/assets/media/blog-images/2022-10-28-multiple-data-source/evolution.png){: .img-fluid}
+
+Multi data source feature introduces a new saved object data-source to describe data source connection, including endpoint, authentication information, etc. Users can dynamically add data source in OpenSearch Dashboards through UI or API.
+
+When using this feature, index pattern which are associated with data source will have reference field filled with:
+
+![IndexPattern with dataSource](/assets/media/blog-images/2022-10-28-multiple-data-source/indexpattern-with-ds.png){: .img-fluid}
+
+Current supported data source saved object attributes (pseudo code):
+
+```
+DataSourceAttributes extends SavedObjectAttributes {
+    title,
+    description,
+    endpoint,
+    auth: {
+      type,
+      credentials,
+    };
+}
+```
+
+In supporting other OpenSearch Dashboards’ components to interact with user data source cluster, this feature forked and repurposed existing `opensearch_service` to allow making a data source connection. When OpenSearch Dashboards needs to access data sources on behalf of the user, it will fetch the data source info from its metadata storage by data source id, then send the request to the corresponding data source endpoint.
+
+How to use the new data source client:
+```ts
+client: OpenSearchClient = await context.dataSource.opensearch.getClient(dataSourceId);
+//Support for legacy client
+apiCaller: LegacyAPICaller = context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI;
+```
+In this phase of development, several OpenSearch Dashboards’ plugins are integrated with the data source concept,  including data, index pattern management, discover and visualization, to allow users to specify a data source when using their functions.
+
+
+**High-level sequence diagram**
+
+![High level sequence diagram](/assets/media/blog-images/2022-10-28-multiple-data-source/highlevel-sequence.png){: .img-fluid}
+
+We plan to provide a more specific design breakdown commit to the OpenSearch Dashboards’ repo. 
```

Comment:
Link the reader to the Dashboards repo.

Suggested rewrite:

A comprehensive design breakdown will be added to the [Dashboards repository](https://github.com/opensearch-project/OpenSearch-Dashboards).